trace-archiver-hdf5/src/main.rs

```diff
@@ -71,9 +81,22 @@ async fn main() -> Result<()> {
     .set("enable.auto.commit", "false")
     .create()?;
 
-    consumer.subscribe(&[&args.trace_topic])?;
+    let topics_to_subscribe = [
+        args.control_topic.as_deref(),
+        Some(args.trace_topic.as_str()),
+    ]
+    .into_iter()
+    .flatten()
+    .collect::<Vec<&str>>();
 
-    let mut trace_file = TraceFile::create(&args.file, args.digitizer_count)?;
+    consumer.subscribe(&topics_to_subscribe)?;
+
+    let mut file = match args.control_topic {
+        // If a control topic is provided, a file will be created for each run.
+        Some(_) => None,
+        // If a a control topic is not provided, a persistent file is used instead.
+        None => Some(TraceFile::create(&args.file, args.digitizer_count)?),
```

Comment:
What does `args.file` do when a control topic is provided?

Reply:
Currently nothing, which is a problem since it is required regardless of what "mode" the tool is running in. I am trying to find a nice way to only make this required if a control topic is not provided.